Math/NumberTheory/Utils/RootsOfUnity.hs

```diff
@@ -0,0 +1,59 @@
+module Math.NumberTheory.Utils.RootsOfUnity
```

Comment:
1. Please copy haddock header from `Math.NumberTheory.DirichletCharacters` and update it appropriately.
2. I would rather go for `Math.NumberTheory.RootsOfUnity`, especially since it needs to be exposed.